openshift-pipelines/tasks/main.yml

```diff
@@ -0,0 +1,266 @@
+            runAfter:
+              - apply-manifests
+
+# Create the build-deploy-api-pipelinerun and build-deploy-ui-pipelinerun Tekton pipelineruns in the pipelines-tutorial namespace
+- name: Create the build-deploy-api-pipelinerun in the pipelines-tutorial namespace
+  kubernetes.core.k8s:
+    state: present
+    api_version: tekton.dev/v1beta1
+    kind: PipelineRun
+    namespace: pipelines-tutorial
+    name: build-deploy-api-pipelinerun
+    definition:
+      apiVersion: tekton.dev/v1
+      kind: PipelineRun
+      metadata:
+        name: build-deploy-api-pipelinerun
+      spec:
+        pipelineRef:
+          name: build-and-deploy
+        taskRunTemplate:
+          serviceAccountName: pipeline
+        params:
+          - name: deployment-name
+            value: pipelines-vote-api
+          - name: git-url
+            value: https://github.com/jianrongzhang89/pipelines-vote-api.git
+          - name: IMAGE
+            value: image-registry.openshift-image-registry.svc:5000/$(context.pipelineRun.namespace)/pipelines-vote-api
+        workspaces:
+          - name: shared-workspace
+            volumeClaimTemplate:
+              spec:
+                accessModes:
+                  - ReadWriteOnce
+                resources:
+                  requests:
+                    storage: 100M
+    wait: true
+    wait_timeout: 90
+    wait_condition:
+      type: Succeeded
+      status: "True"
+      reason: Succeeded
+
+- name: Create the build-deploy-ui-pipelinerun in the pipelines-tutorial namespace
+  kubernetes.core.k8s:
+    state: present
+    api_version: tekton.dev/v1beta1
+    kind: PipelineRun
+    namespace: pipelines-tutorial
+    name: build-deploy-ui-pipelinerun
+    definition:
+      apiVersion: tekton.dev/v1
+      kind: PipelineRun
+      metadata:
+        name: build-deploy-ui-pipelinerun
+      spec:
+        pipelineRef:
+          name: build-and-deploy
+        params:
+          - name: deployment-name
+            value: pipelines-vote-ui
+          - name: git-url
+            value: https://github.com/jianrongzhang89/pipelines-vote-ui.git
+          - name: IMAGE
+            value: image-registry.openshift-image-registry.svc:5000/$(context.pipelineRun.namespace)/pipelines-vote-ui
+        workspaces:
+          - name: shared-workspace
+            volumeClaimTemplate:
+              spec:
+                accessModes:
+                  - ReadWriteOnce
+                resources:
+                  requests:
+                    storage: 100Mi
+    wait: true
+    wait_timeout: 90
+    wait_condition:
+      type: Succeeded
+      status: "True"
+      reason: Succeeded
+
+# Remove the pipelines-tutorial namespace; will remove all the resources created in the namespace
+- name: Remove the pipelines-tutorial namespace
+  kubernetes.core.k8s:
+    state: absent
+    api_version: v1
+    kind: Namespace
+    name: pipelines-tutorial
+    wait: true
+    wait_timeout: 60
+    wait_condition:
```

Comment:
This isn't strictly necessary. For state: absent, and wait: true it automatically does the right thing, and waits until it is actually deleted.